filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsFileSystemProvider.scala

```diff
@@ -201,7 +203,8 @@ class GcsFileSystemProvider private (storageClient: Try[Storage], executionConte
   override def isHidden(path: Path): Boolean = throw new NotImplementedError()
 
   private def list(gcsDir: NioGcsPath) = {
-    val listRequest = client.objects().list(gcsDir.bucket)
+    val maxResults = ConfigFactory.load().getIntOr("google.list-max-results", 50000).toLong
+    val listRequest = client.objects().list(gcsDir.bucket).setMaxResults(maxResults)
```

Comment:
Have you verified that google will actually honour such a high number? 

Do you need to do an equivalent fix on `glob()`?